src/Bicep.Core/Modules/OciArtifactModuleReference.cs

```diff
@@ -73,35 +81,78 @@ public override int GetHashCode()
 
         public static OciArtifactModuleReference? TryParse(string rawValue, out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder)
         {
-            static DiagnosticBuilder.ErrorBuilderDelegate CreateErrorFunc(string rawValue) => x => x.InvalidOciArtifactReference($"oci:{rawValue}");
+            static DiagnosticBuilder.ErrorBuilderDelegate CreateErrorFunc(string rawValue, string? innerError) => x => x.InvalidOciArtifactReference($"{ModuleReferenceSchemes.Oci}:{rawValue}", innerError);
+
+            static DiagnosticBuilder.ErrorBuilderDelegate CreateInvalidSegmentErrorFunc(string rawValue, string badSegment) =>
+                CreateErrorFunc(
+                    rawValue,
+                    $"The mpdule path segment \"{badSegment}\" is not valid. Each module name path segment must be a lowercase alphanumeric string optionally separated by a \".\", \"_\" , or \"-\".");
```

Comment:
typo ("mpdule"). Should we move this function into `DiagnosticBuilder` for the sake of grouping error strings in one place?

Reply:
Good pt. I fixed the typo and moved them over to `DiagnosticBuilder`.